app/services/processors/jira_project_defect_escape_rate_updater.rb

```diff
@@ -27,17 +30,34 @@ def bugs_to_update
       JiraClient::Repository.new(@jira_board).bugs
     end
 
-    def issue_update!(issue, bug_fields)
+    def issue_update!(issue, bug_fields, histories)
       environment_field = bug_fields[JIRA_ENVIRONMENT_FIELD]
-      status_field = bug_fields[:status][:name]
+
+      in_progress_at = in_progress_at(histories)
 
       issue.update!(
         informed_at: bug_fields[:created],
         resolved_at: bug_fields[:resolutiondate] || nil,
-        in_progress_at: status_field == IN_PROGRESS ? bug_fields[:updated] : issue.in_progress_at,
+        in_progress_at: in_progress_at.presence || issue.in_progress_at,
         environment: environment_field ? environment_field.first[:value].downcase : nil,
         issue_type: 'bug'
       )
     end
+
+    def in_progress_at(histories)
+      return unless histories
+
+      @in_progress_at = nil
+      histories.each do |history|
+        items = history[:items]&.first
+
+        if items && items[:fieldId] == 'status' && items[:fromString] == 'To Do' &&
```

Comment:
Maybe it is useful to standardize the use of constants that are fixed strings of Jira environment. 

`'In Progress'`
`'To Do'`
`'status'`

WDYT?

Reply:
Done!